agent/api/task/task.go

```diff
@@ -2067,6 +2158,12 @@ func (task *Task) PopulateSecrets(hostConfig *dockercontainer.HostConfig, contai
 		asmRes = resource[0].(*asmsecret.ASMSecretResource)
 	}
 
+	populateContainerSecrets(hostConfig, container, ssmRes, asmRes)
+	return nil
+}
+
+func populateContainerSecrets(hostConfig *dockercontainer.HostConfig, container *apicontainer.Container,
```

Comment:
why this is separated to another function. I don't feel it is necessary here.

Reply:
it exceeds the complexity limit of the gocyclo target if i don't do this separation